src/layers/core/polygon-layer/polygon-layer.js

```diff
@@ -83,26 +83,26 @@ export default class PolygonLayer extends CompositeLayer {
     const {data, id, stroked, filled, extruded, wireframe} = this.props;
     const {paths, onHover, onClick} = this.state;
 
-    const strokePolygons = stroked && data && data.length > 0;
-    const fillPolygons = filled && data && data.length > 0;
+    const polygon = data && data.length > 0;
 
     // Filled Polygon Layer
-    const polygonFillLayer = fillPolygons && new SolidPolygonLayer(Object.assign({},
+    const polygonFillLayer = filled && polygon && new SolidPolygonLayer(Object.assign({},
       this.props, {
         id: `${id}-fill`,
         data,
         getElevation,
         getColor: getFillColor,
         extruded,
-        wireframe,
-        updateTriggers: Object.assign({}, updateTriggers, {
+        wireframe: false,
+        updateTriggers: {
+          getElevation: updateTriggers.getElevation,
           getColor: updateTriggers.getFillColor
-        })
+        }
       }));
 
     // Polygon outline layer
     let polygonOutlineLayer = null;
-    if (strokePolygons) {
+    if (!extruded && stroked && polygon) {
```

Comment:
Here `polygon` has only an indirect relation to `paths`. Could be better to check `paths` directly.